### DISCUSSION TITLE
how to speed up the compilation?

### DISCUSSION THREAD

**Discussion by whisperingkahuna** [sjtug/SJTUThesis#832] [提问] [answered]
I am using this template for a 130-page document. It takes approximately one minute to fully compile. Any modifications to make it faster?
below is the output.
```
Process started: latexmk.exe -xelatex -silent -synctex=1 "main"

Latexmk: Run number 1 of rule 'xelatex'
Latexmk: Run number 2 of rule 'xelatex'
Latexmk: Run number 1 of rule 'xdvipdfmx'
Process exited normally
```

Regards,

**Comment by LogCreative** [comment, marked as answer]
There are mainly two directions to make the compilation faster:
 - One way is to disable some of the chapters (`\input{...}`) and compile the shorter document when you are writing. Compile the whole document when you finish it. If you want to keep track of the cross references, use `\include{...}` instead of `\input{...}`, refer to [LearnLaTeX - Structuring Longer Documents](https://www.learnlatex.org/en/lesson-13) for more details.
 - The other way is to really make the compilation faster. 
a) If you use the `tikz` package for drawing some diagrams, try to pre-compile the diagrams in a [`standalone` document class](https://ctan.math.illinois.edu/macros/latex/contrib/standalone/standalone.pdf#5) and use `\includegraphics{...}` in your thesis. Or try to use [`externalize` tikz package](https://mirrors.sjtug.sjtu.edu.cn/CTAN/graphics/pgf/base/doc/pgfmanual.pdf#52) for automated pre-compilation. If you are not using `tikz` in your thesis, remove the package loading in
 https://github.com/sjtug/SJTUThesis/blob/864c1551d049e0b84185c282df64e891be3a2056/setup.tex#L178-L179
b) If you inserted a lot of pictures, try to pass the `draft` option to the document class. When the writing is finished, remove the option for the final "release".
https://github.com/sjtug/SJTUThesis/blob/864c1551d049e0b84185c282df64e891be3a2056/main.tex#L4
c) If you only want to inspect the layout of your thesis, you could manually decrease the number of compilation times. In this way, the cross-references could be wrong but the single-time compilation is still 1/3 of the time.
d) If you are using Windows, try to use [Windows Subsystem for Linux (WSL)](https://docs.microsoft.com/en-us/windows/wsl/install), you could get a much faster speed thanks to a faster file system.

**Reply by whisperingkahuna** [reply]
I really appreciate your detailed response.

Some confusion with c and d
How do I decrease the number of compilation times?
I have installed WSL, now, I am wondering what changes should I make in texstudio in order to use WSL during compilation.

Regards,

**Reply by LogCreative** [reply]
c) means you could just input the command like
```cmd
xelatex main
biber main
xelatex main
xelatex main
```
manually, but it is not recommended, since it is not the final result in the intermediate steps.

d) I know that using Visual Studio Code is very convenient, just open up the remote server via a plugin. As for the texstudio, you probably need to search for some solutions.

**Reply by whisperingkahuna** [reply]
I tried all of these ways, options b seems to be the best option for me. 

Thank you for your help
